bayesian/utility/v2/ca_response_builder.py

```diff
@@ -76,8 +116,78 @@ def get_component_analyses_response(self):
             return None
 
 
-class ComponentAnalysisResponseBuilder:
-    """Vendor specific response builder for Component Analyses v2."""
+class CABatchCall:
+    """Namespace for Component Analyses Batch call."""
+
+    def __init__(self, ecosystem: str, packages: List[Dict]):
+        """For Flows related to Security Vendor Integrations."""
+        self.ecosystem = ecosystem
+        self.packages = packages
+        self.recommendation = dict()
+        self.response_data = dict()
+
+    def get_ca_batch_response(self) -> Union[Tuple, None]:
+        """Fetch analysis for given package+version from the graph database.
+
+        Build CA Batch Response.
+        Result is fed to ResponseBuilder.
+        :param packages: List of dict of package, version info.
+
+        :returns:
+            - Json Response
+            - None: Exception/Package not Known.
+        """
+        logger.info('Executing CA Batch Vendor Specific Analyses')
+        try:
+            graph_response = GraphAnalyses().get_batch_ca_data(
+                self.ecosystem, self.packages, 'ca_batch')
+
+            analyzed_dependencies = set(self._analysed_package_details(graph_response))
+            unknown_pkgs: Set = self.get_all_unknown_packages(analyzed_dependencies)
+            result = []
+            for package in analyzed_dependencies:
+                result.append(CABatchResponseBuilder(
+                    self.ecosystem, package.name, package.version).generate_recommendation(
+                    graph_response))
+            return result, unknown_pkgs
+
+        except Exception as e:
+            logger.info("Error")
+            logger.error(str(e))
+            return None, None
+
+    @staticmethod
+    def _analysed_package_details(graph_response: Dict) -> Set:
+        """Analyses Package Details from GraphDB.
+
+        Converts GraphDb output packages into list of Normalised Packages
+        :param graph_response: Graph DB Response
+        :return: list of Normalised Packages
+        """
+        db_pkg_list = []
+        for pack_details in graph_response.get('result').get('data'):
+            pkg_name = pack_details.get('package').get('name', [''])[0]
+            pkg_vr = pack_details.get('version').get('version', [''])[0]
+            db_pkg_list.append({"name": pkg_name, "version": pkg_vr})
+        _normalized_packages = NormalizedPackages(db_pkg_list)
```

Comment:
The code in NormalizedPackages can be written here

Reply:
Changed `NormalizedPackages` class to `normlize_packages` method.